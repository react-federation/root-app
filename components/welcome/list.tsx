import { Text } from '@chakra-ui/layout'
import React from 'react'
import DynamicList from './dynamic-list'

const createVscodeUrl = (path: string) => `vscode://file${PROJECT_PATH}${path}`

export const styling = [
  {
    href: createVscodeUrl('/components/html-wrapper/index.tsx'),
    children: 'CSS-in-JS (We prefer to use this one!)'
  },
  {
    href: createVscodeUrl('/components/welcome/welcome.module.scss'),
    children: 'CSS modules'
  },
  {
    href: createVscodeUrl('/styles/main.scss'),
    children: 'Classic CSS/SCSS'
  }
]

export const list = [
  {
    children: (
      <>
        <Text>Styling</Text>
        <DynamicList items={styling} />
      </>
    )
  },
  { children: 'How to write components' },
  { children: 'Data fetching' },
  { children: 'Global state' },
  { children: 'Routing' },
  { children: 'Testing' }
]
