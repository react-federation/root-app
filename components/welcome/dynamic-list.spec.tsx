import setup from 'utils/test/setup'
import DynamicList from './dynamic-list'

import { list } from './list'

describe('DynamicList', () => {
  it('should match snapshot', () => {
    const { asFragment } = setup(DynamicList, { items: list })

    expect(asFragment()).toMatchSnapshot()
  })
})
