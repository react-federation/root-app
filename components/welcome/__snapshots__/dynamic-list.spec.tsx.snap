// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicList should match snapshot 1`] = `
<DocumentFragment>
  <ul
    class="list-style"
  >
    <li>
      <p
        class="chakra-text emotion-0"
      >
        Styling
      </p>
      <ul
        class="list-style"
      >
        <a
          href="vscode://file//components/html-wrapper/index.tsx"
        >
          <li>
            CSS-in-JS (We prefer to use this one!)
          </li>
        </a>
        <a
          href="vscode://file//components/welcome/welcome.module.scss"
        >
          <li>
            CSS modules
          </li>
        </a>
        <a
          href="vscode://file//styles/main.scss"
        >
          <li>
            Classic CSS/SCSS
          </li>
        </a>
      </ul>
    </li>
    <li>
      How to write components
    </li>
    <li>
      Data fetching
    </li>
    <li>
      Global state
    </li>
    <li>
      Routing
    </li>
    <li>
      Testing
    </li>
  </ul>
</DocumentFragment>
`;
